system/library/razorpay-lib/createwebhook.php

```diff
@@ -0,0 +1,176 @@
+
+            return $this->returnWebhookConfigData();
+        }
+
+        try
+        {
+            $webhookPresent = $this->getExistingWebhook();
+
+            if(empty($this->webhookId) === false)
+            {
+                $webhook = $api->webhook->edit(
+                    [
+                        'url'    => $this->webhookUrl,
+                        'events' => $this->webhookEvents,
+                        'secret' => $this->webhookSecret,
+                        'active' => true,
+                    ],
+                    $this->webhookId
+                );
+
+                $this->log->write('Razorpay Webhook Updated by Admin.');
+            }
+            else
+            {
+                $webhook = $api->webhook->create(
+                    [
+                        'url'    => $this->webhookUrl,
+                        'events' => $this->webhookEvents,
+                        'secret' => $this->webhookSecret,
+                        'active' => true,
+                    ]
+                );
+
+                $this->log->write('Razorpay Webhook Created by Admin');
+            }
+
+            $this->webhookEnable = '1';
+
+            return $this->returnWebhookConfigData();
+        }
+        catch(\Razorpay\Api\Errors\Error $e)
+        {
+            $this->webhookEnable = '0';
+            $this->log->write($e->getMessage());
+
+            return $this->returnWebhookConfigData();
+        }
+    }
+
+    protected function getExistingWebhook()
+    {
+        $api = $this->getApiIntance();
+
+        try
+        {
+            $webhooks = $api->webhook->all();
+
+            if(($webhooks->count > 0) and
+                (empty($this->webhookUrl) === false))
+            {
+                foreach ($webhooks->items as $key => $webhook)
+                {
+                    if($webhook->url === $this->webhookUrl)
+                    {
+                        $this->webhookId = $webhook->id;
+
+                        foreach ($webhook->events as $event => $status)
+                        {
+                            if(($status === true) and
+                                (in_array($event, $this->webhookSupportedEvents)) === true)
+                            {
+                                $this->webhookEvents[$event] = true;
+                            }
+                        }
+
+                        return ['id' => $webhook->id];
+                    }
+                }
+            }
+        }
+        catch(\Razorpay\Api\Errors\Error $e)
+        {
+            $this->webhookEnable = '0';
+            $this->log->write($e->getMessage());
+
+            return ['id' => null];
+        }
+
+        return ['id' => null];
+    }
+
+    protected function createWebhookSecret()
+    {
+        $alphanumericString = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
```

Comment:
@abdulwahidsharief , let's add some supported special chars list as well. Also increase the length to 20 chars.
$alphanumericString = '0123456789#@$^*ABCDEFGHIJKLMNOPQRSTUVWXYZ#@$^*abcdefghijklmnopqrstuvwxyz0123456789';

Reply:
@ramth05 included special chars and limit is set to 20